pkg/project/admission/nodeenv/admission.go

```diff
@@ -52,6 +57,16 @@ func (p *podNodeEnvironment) Admit(a admission.Attributes) (err error) {
 	if !p.cache.Running() {
 		return err
 	}
+
+	// if this is a pod being created by a daemonset we do not want to set the node selector automatically
+	isDaemonset, err := p.isDaemonsetPod(pod)
+	if err != nil {
+		return apierrors.NewForbidden(resource, name, fmt.Errorf("an unexpected error occured when checking for daemonsets: %v", err))
+	}
+	if isDaemonset {
+		return nil
```

Comment:
I double check in the isDaemonsetPod that the pod would actually fall under the selector of the DS.  So if you fake it you have to fake it so that it would fall under the DS anyway